src/aws/ec2Instances.ts

```diff
@@ -1,21 +1,34 @@
-import {describeAsg} from './autoscaling'
-import {Instance} from './types'
-import {AutoScalingGroupsType} from 'aws-sdk/clients/autoscaling';
-import {DescribeInstancesResult, TerminateInstancesResult} from 'aws-sdk/clients/ec2';
+import {Instance} from './types';
+import {DescribeInstancesResult, Instance as EC2Instance} from 'aws-sdk/clients/ec2';
 
 const AWS = require('aws-sdk');
 const awsEc2 = new AWS.EC2();
 
 type InstanceFilter = (instances: Instance[]) => Instance;
 
+function buildInstance(instance: EC2Instance): Instance {
+    const autoScalingGroupNameTag = instance.Tags!.find(({ Key }) => Key === "aws:autoscaling:groupName");
```

Comment:
We could also use the [DescribeAutoScalingInstances](https://docs.aws.amazon.com/cli/latest/reference/autoscaling/describe-auto-scaling-instances.html) API to do this but as far as I know it's safe to rely on that tag, apart from the general eventual consistency issues inherent to tags.